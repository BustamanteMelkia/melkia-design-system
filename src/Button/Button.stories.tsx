import React from "react";
import { type Meta, type Story } from "@storybook/react";
import Button, { type Props } from "./Button";

const meta: Meta = {
  title: "Button",
  component: Button,
};

export default meta;

const Template: Story<Props> = (args) => <Button {...args} />;

export const Default = Template.bind({});
Default.args = {};

export const Secondary = Template.bind({});
Secondary.args = {
  variant: "secondary",
};
