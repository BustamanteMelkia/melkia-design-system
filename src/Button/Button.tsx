import React, { type ReactNode, type HTMLAttributes } from "react";
import { createUseStyles } from "react-jss";

interface StylesProps {
  size: "sm" | "md" | "lg";
  variant: "primary" | "secondary" | "danger";
}
interface BaseProps extends HTMLAttributes<HTMLButtonElement> {
  children: ReactNode;
}
export type Props = StylesProps & BaseProps;

const useStyles = createUseStyles({
  button: ({ variant }: StylesProps) => ({
    backgroundColor: "blue",
    color: "white",
    border: "none",
    padding: "8px 12px",
    borderRadius: "8px",
  }),
});

const Button = ({
  children = "Click this!",
  size = "md",
  variant = "primary",
  ...props
}: Props): JSX.Element => {
  const classes = useStyles({ variant, size });
  return (
    <button {...props} className={classes.button}>
      {children}
    </button>
  );
};

export default Button;
