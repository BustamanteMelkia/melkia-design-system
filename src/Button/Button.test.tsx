import React from "react";
import * as ReactTestRenderer from "react-test-renderer";
import { expect, it } from "@jest/globals";
import Button from "./Button";

it("Matches DOM Snapshot", () => {
  const tree = ReactTestRenderer.create(
    <Button size="md" variant="primary">
      hello
    </Button>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
