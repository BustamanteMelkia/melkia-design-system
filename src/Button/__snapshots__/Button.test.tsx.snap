// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches DOM Snapshot 1`] = `
<button
  className="button-0-2-1 button-d0-0-2-2"
>
  hello
</button>
`;
